test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -0,0 +1,352 @@
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # Submit a high priority job
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:mem=512mb',
+             ATTR_q: 'expressq'})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S', 'substate': 45}, id=jid1)
+
+        job = self.server.status(JOB, id=jid1)
+        self.assertNotEqual(job, None)
+
+        rr = "(%s:ncpus=4)" % self.mom.shortname
+        self.assertEqual(job[0][ATTR_released], rr,
+                         msg="resources_released incorrect")
+
+    def test_res_released_list_sched_susp(self):
+        """
+        Test if job's resources_released_list attribute is correctly set when
+        it is suspended.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus,mem'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # Submit a high priority job
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:mem=512mb',
+             ATTR_q: 'expressq'})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S', 'substate': 45}, id=jid1)
+
+        job = self.server.status(JOB, id=jid1)
+        self.assertNotEqual(job, None)
+
+        rr_l_ncpus = job[0][ATTR_rel_l + ".ncpus"]
+        self.assertEqual(rr_l_ncpus, "4", msg="ncpus not released")
+        rr_l_mem = job[0][ATTR_rel_l + ".mem"]
+        self.assertEqual(rr_l_mem, "524288kb", msg="memory not released")
+
+    def test_node_res_after_deleting_suspended_job(self):
+        """
+        Test that once a suspended job is deleted node's resources assigned
+        are back to 0.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        rc = 0
+        try:
+            rc = self.server.sigjob(jobid=jid1, signal="suspend")
+        except PbsSignalError, e:
+            self.assertEqual(rc, 0, e.msg[0])
+
+        time.sleep(1)
```

Comment:
Why are you sleeping?  Are you waiting for the suspend to go through?  If so, you should use expect().  Sleeping will cause a race condition.